prospector/datamodel/advisory.py

```diff
@@ -123,15 +123,28 @@ def extract_path_tokens(text: str) -> "list[str]":
     Returns:
         list: a list of paths that are found
     """
-    return [
-        re.split(r"\.|,|/", token.rstrip(r".,;:?!\"'"))
-        for token in text.split(" ")
-        if ("/" in token.rstrip(r".,;:?!\"'") and not token.startswith("</"))
-        or (
-            "." in token.rstrip(r".,;:?!\"'")
-            and token.rstrip(r".,;:?!\"'").split(".")[-1] in RELEVANT_EXTENSIONS
+    tokens = re.split(r"\s+", text)  # split the text into words
+    tokens = [
+        token.strip(",.:;-+!?)]}'\"") for token in tokens
+    ]  # removing common punctuation marks
+    paths = []
+    for token in tokens:
+        is_contains_path_separators = ("\\" in token) or ("/" in token)
+        is_period_separated = "." in token
+        has_relevant_extension = token.split(".")[-1] in RELEVANT_EXTENSIONS
+        is_xml_tag = token.startswith("<")
+        is_property = token.endswith("=")
+
+        # if strict_extensions:
+        # it will always match tokens with (back) slashes,
+        # but it will only match single file names if they have the correct extension
+        is_path_like = is_contains_path_separators or (
+            has_relevant_extension if strict_extensions else is_period_separated
         )
-    ]
+        is_path_dislike = is_xml_tag or is_property
```

Comment:
I am not sure I understand this....

Reply:
Maybe someone mention an XML tag like <simpletag> or some properties key=value in the comment text. My assumption was that we do not want to report these as paths.

Reply:
What i would do (to simplify things but also to leave us with more options) is to treat all strings that are not matching `[a-z]+` as tokens that could be potentially interesting for matching against paths or other commit features...

We can then try to "categorize" those tokens, but the categorization is sort of good-to-have (arguably, even just collecting them would be powerful enough for most purposes)

Reply:
I understand this, but then should I merge the functionality of the path and interesting token extraction logic? OR should this PR be kept just for refactoring and fixing the path extraction return type? I vote for the last one since we strive for simple PRs.

Reply:
I agree, let's wrap this up with the minimal changes required to move forward.